webui/src/pages/repositories/repository/objects.jsx

```diff
@@ -3,22 +3,232 @@ import React, {useRef, useState} from "react";
+    };
+
+    const hide = () => {
+        if (importState.inProgress) return;
+        setImportState(initialState)
+        onHide()
+    };
+
+    const ingest = async () => {
+        setImportState({
+            ...initialState,
+            inProgress: true
+        })
+
+        try {
+            let hasMore = true
+            let continuationToken = ""
+            let after = ""
+            let prepend = `${destRef.current.value}`;
+            let importBranchResp
+            let sum = importState.numObj
+            const importBranch = `${reference.id}-imported`;
+            const commitMsg = commitMsgRef.current.value;
+            const sourceRefVal = sourceRef.current.value;
+            const range_arr = []
+
+            while (hasMore) {
+                const response = await ranges.createRange(repo.id, sourceRefVal, after, prepend, continuationToken)
+                range_arr.push(response.range)
+
+                let pagination = response.pagination
+                hasMore = pagination.has_more
+                continuationToken = pagination.continuation_token
+                after = pagination.last_key
+                sum += response.range.count
+                setImportState({inProgress: true, error: null, done: false, isStorageNamespaceValid: importState.isStorageNamespaceValid, numObj: sum})
+            }
+            const metarange = await metaRanges.createMetaRange(repo.id, range_arr)
 
-const UploadButton = ({ config, repo, reference, path, onDone, variant = "success", onClick, onHide, show = false}) => {
+            try {
+                importBranchResp = await branches.get(repo.id, importBranch)
+            } catch (error){
+                await branches.create(repo.id, importBranch, reference.id)
+                importBranchResp = await branches.get(repo.id, importBranch)
+            }
+
+            await commits.commit(repo.id, importBranchResp.id, commitMsg, {}, metarange.id)
+            setImportState({inProgress: false, error: null, done: true, numObj: sum, isStorageNamespaceValid: importState.isStorageNamespaceValid})
+            onDone()
+        } catch (error) {
+            setImportState({...initialState, error})
+            throw error
+        }
+    }
+    const basePath = `lakefs://${repo.id}/${reference.id}-imported/\u00A0`;
+    const pathStyle = {'minWidth' : '25%'};
+
+    return (
+        <>
+            <Modal show={show} onHide={hide} size="lg">
+                <Modal.Header closeButton>
+                    <Modal.Title>Import</Modal.Title>
+                </Modal.Header>
+                {(importState.done) ? (<ImportDone currBranch={reference.id} importBranch={`${reference.id}-imported`} numObjects={importState.numObj} />)
+                    : ( <>
+                <Modal.Body>
+                {(importState.inProgress) ? (<ImportProgress numObjects={importState.numObj} />)
+                    : ( <>
+                            <Alert variant="warning">
```

Comment:
That's a good thing! Make it "info" or something happier

Reply:
Changed to info